src/Compiler/pars.fsy

```diff
@@ -4663,13 +4666,29 @@ arrayExpr:
       { reportParseErrorAt (rhs parseState 1) (FSComp.SR.parsUnmatchedBracketBar())
         exprFromParseError ($2 (rhs2 parseState 1 2)) }
 
-  | LBRACK_BAR error BAR_RBRACK
-      { (* silent recovery *) SynExpr.ArrayOrList(true, [ ], lhs parseState) }
+  | LBRACK_BAR error BAR_RBRACK 
+      {  (* silent recovery *) SynExpr.ArrayOrList (CollectionType.Array, [ ], lhs parseState) }  
 
   | LBRACK_BAR recover
-      { reportParseErrorAt (rhs parseState 1) (FSComp.SR.parsUnmatchedBracketBar())
+      { reportParseErrorAt (rhs parseState 1) (FSComp.SR.parsUnmatchedBracketBar())  
+        (* silent recovery *) 
+        exprFromParseError (SynExpr.ArrayOrList (CollectionType.Array, [ ], rhs parseState 1)) }  
+
+immarrayExpr:
+  | LBRACK_COLON immarrayExprElements COLON_RBRACK
+      {  $2 (lhs parseState) }
+
+  | LBRACK_COLON immarrayExprElements recover
+      { reportParseErrorAt (rhs parseState 1) (FSComp.SR.parsUnmatchedBracketBar()) // Don't know where this is, can't change to BracketColon
```

Comment:
Don't know where this is, can't change to BracketColon

Reply:
I'm a bit confused about what your question is here.
What can't you find? What can you not change because for what reason?

Reply:
The question is how to do the appropriate analogy to write the immarray code based on the array code. I don't understand the array code: the term `FSComp.SR.parsUnmatchedBracketBar()`, appearing in an fsy file, looks similar to calling a function, but the function doesn't seem to have a definition. Perhaps the definition is automatically generated?

Reply:
Those are defined in the fscomp text file and later code-generated

Reply:
thanks and sorry I should have figured that out!